src/main/scala/net/ripe/rpki/publicationserver/repository/HashingSizedStream.scala

```diff
@@ -1,23 +1,33 @@
 package net.ripe.rpki.publicationserver.repository
 
-import java.io.OutputStream
-import java.security.MessageDigest
+import net.ripe.rpki.publicationserver.Binaries.Bytes
+import net.ripe.rpki.publicationserver.Hash
 
-import net.ripe.rpki.publicationserver.{Hash, Hashing}
+import java.io.{FilterOutputStream, OutputStream}
+import java.security.MessageDigest
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
 
-class HashingSizedStream(val os: OutputStream) extends Hashing {
+class HashingSizedStream(secret: Bytes, os: OutputStream) extends FilterOutputStream(os) {
   private var size: Long = 0
   private val digest = MessageDigest.getInstance("SHA-256")
+  private val mac = Mac.getInstance("HmacSHA256")
 
-  def write(bytes: Array[Byte]): Unit = {
-    digest.update(bytes)
-    size += bytes.length
-    os.write(bytes)
-  }
+  mac.init(new SecretKeySpec(secret.value, "HmacSHA256"))
 
-  def summary = (Hash(digest.digest()), size)
+  override def write(b: Int): Unit = {
+    out.write(b)
+    digest.update(b.asInstanceOf[Byte])
+    mac.update(b.asInstanceOf[Byte])
+    size += 1
+  }
 
-  def flush() = os.flush()
+  override def write(bytes: Array[Byte], offset: Int, len: Int): Unit = {
+    out.write(bytes, offset, len)
+    digest.update(bytes, offset, len)
+    mac.update(bytes, offset, len)
```

Comment:
`out.write(bytes, offset, len)` and `digest.update` and `mac.update` are symmetrical w.r.t. behaviour -> this works.